chessington/engine/pieces.py

```diff
@@ -33,14 +34,90 @@ class Pawn(Piece):
+            if board.get_piece(Square.at(current_square.row - distance, current_square.col)):
+                return True
+        else:
+            if board.get_piece(Square.at(current_square.row + distance, current_square.col)):
+                return True
+
+        return False
+
+    def at_edge(self, square: Square) -> bool:
+        if self.player == Player.BLACK:
+            if square.row == 0:
+                return True
+        else:
+            if square.row == 7:
+                return True
+
+        return False
+
+    def has_piece_diagonal(self, board: Board, square: Square) -> bool:
+        if self.player == Player.BLACK:
+            diag_left = board.get_piece(Square.at(square.row - 1, square.col - 1))
+            diag_right = board.get_piece(Square.at(square.row - 1, square.col + 1))
+
+            if (diag_left and diag_left.player == Player.WHITE) \
+                    or (diag_right and diag_right.player == Player.BLACK):
+                return True
+        else:
+            diag_left = board.get_piece(Square.at(square.row + 1, square.col - 1))
+            diag_right = board.get_piece(Square.at(square.row + 1, square.col + 1))
+
+            if (diag_left and diag_left.player == Player.BLACK) \
+                    or (diag_right and diag_right.player == Player.BLACK):
+                return True
+        return False
+
+    def get_moves_diag(self, board: Board, square: Square) -> List[Square]:
+        square_list = []
+
         if self.player == Player.BLACK:
-            square_in_front = Square.at(current_square.row - 1, current_square.col)
-            return [square_in_front]
+            diag_left = board.get_piece(Square.at(square.row - 1, square.col - 1))
+            diag_right = board.get_piece(Square.at(square.row - 1, square.col + 1))
+
+            if diag_left and diag_left.player == Player.WHITE:
+                square_list.append(Square.at(square.row - 1, square.col - 1))
+            if diag_right and diag_right.player == Player.WHITE:
+                square_list.append(Square.at(square.row - 1, square.col + 1))
         else:
-            square_in_front = Square.at(current_square.row + 1, current_square.col)
-            return [square_in_front]
+            diag_left = board.get_piece(Square.at(square.row + 1, square.col - 1))
+            diag_right = board.get_piece(Square.at(square.row + 1, square.col + 1))
+
+            if diag_left and diag_left.player == Player.BLACK:
+                square_list.append(Square.at(square.row + 1, square.col - 1))
+            if diag_right and diag_right.player == Player.BLACK:
+                square_list.append(Square.at(square.row + 1, square.col + 1))
+
+        return square_list
+
+    def can_move(self, board: Board, square: Square) -> bool:
+        if not self.at_edge(square):
+            return self.has_piece_diagonal(board, square) or not self.has_piece_in_front(board, square)
+
+        return False
 
```

Comment:
I have a small preference for having the auxiliary functions written above the function where they are used, can you move them above, please? 😄